### PR TITLE
[release/6.0] Backport 48892 to 6.0

### PR DESCRIPTION
**Pull request by mitchdenny**
Backports #48892

### CODE REVIEW

**Review by halter73 — APPROVED** [review]

File: src/Middleware/WebSockets/src/ServerWebSocket.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.WebSockets;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.WebSockets;
```
Comment (amcasey) [inline src/Middleware/WebSockets/src/ServerWebSocket.cs:7]: I don't think we used file-level namespace statement (name?) in 6.0.

**Comment by danmoseley** [comment]
approved by tactics

**Comment by mitchdenny** [comment]
Latest PR was to fix up a test case that was flakey (we picked up on main so have applied it to the backport as well so we don't make the servicing branch flakey.

File: src/Middleware/WebSockets/src/ServerWebSocket.cs (hunk 1 above)
Comment (mitchdenny) [reply src/Middleware/WebSockets/src/ServerWebSocket.cs]: Fixed. Thanks.

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**